binzx/otomi

```diff
@@ -41,37 +41,73 @@ silent() {
     "$@" &>/dev/null
   fi
 }
+
+docker_tag_exists() {
+  curl -sflL https://index.docker.io/v1/repositories/$1/tags/$2 &>/dev/null
+}
+
 # shellcheck disable=SC2155
 readonly base_dir="$(cd "$(dirname "${BASH_SOURCE[0]}")" &>/dev/null && pwd)"
 
-if [[ $base_dir == *"otomi-core"* ]] || [[ $(pwd) == *"otomi-core"* ]]; then
+findUp() {
+  path=$1
+  while [[ "$path" != "" && ! -e "$path/$2" ]]; do
+    path=${path%/*}
+  done
+  echo "$path"
+}
+
+packageIsCore() {
+  path=$1
+  [ -z $path ] && return 1
+  packageName=$(grep name "$path/package.json" | sed 's/.*"name": "\(.*\)".*/\1/')
+  [[ $packageName == 'otomi-core' ]] && return 0 || return 1
+}
+
+packageJsonPathPwd=$(findUp $(pwd) 'package.json')
+packageJsonPathBaseDir=$(findUp $base_dir 'package.json')
+
+if packageIsCore $packageJsonPathPwd; then
   in_core=1
+  core_path=$packageJsonPathPwd
+elif packageIsCore $packageJsonPathBaseDir; then
+  in_core=1
+  core_path=$packageJsonPathBaseDir
 fi
+
 [ -z $ENV_DIR ] && [ -n "$in_core" ] && echo "No ENV_DIR set!" && exit 1
 [ -z $ENV_DIR ] && ENV_DIR="$PWD"
 # Parse yaml block without yq: https://stackoverflow.com/a/49190338/14982291
-otomi_version="latest"
+fallback_otomi_version='latest'
+fallback_otomi_dev_version='master'
 if [ -n "$in_core" ]; then
   current_branch=$(git branch --show-current 2>/dev/null)
   current_branch=${current_branch##*/}
-  current_branch=${current_branch:-'master'}
+  current_branch=${current_branch:-$fallback_otomi_dev_version}
   otomi_version=$current_branch
 elif [ -f "${ENV_DIR}/env/settings.yaml" ]; then
   otomi_version=$(awk '/otomi/{flag=1} flag && /version:/{print $NF;flag=""}' $ENV_DIR/env/settings.yaml)
+  otomi_version=${otomi_version:-$fallback_otomi_version}
 fi
 OTOMI_TAG=${OTOMI_TAG:-$otomi_version}
 
+if [ -z $NOPULL ] && ! docker_tag_exists otomi/core $OTOMI_TAG; then
+  [ -n "$in_core" ] && otomi_version=$fallback_otomi_dev_version || otomi_version=$fallback_otomi_version
+  echo "Docker doesn't have otomi/core:${OTOMI_TAG}."
+  echo "Falling back to otomi/core:${otomi_version}"
+  OTOMI_TAG=$otomi_version
+fi
```

Comment:
Please just bail when wrong input is provided. Never just assume what the user wants...

Reply:
When in dev mode, it is necessary, when image doesn't exist yet to fall back because if chicken egg problem